public/app/features/alerting/unified/api/alertingApi.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
+
+import { PromBuildInfoResponse } from '../../../../types/unified-alerting-dto';
+
+import { discoverAlertmanagerFeatures } from './buildInfo';
+
+const backendSrvBaseQuery = (): BaseQueryFn<BackendSrvRequest> => async (requestOptions) => {
+  try {
+    const response = await lastValueFrom(getBackendSrv().fetch<PromBuildInfoResponse>(requestOptions));
+
+    return { data: response.data };
+  } catch (error) {
+    return {
+      error: {
+        data: error,
+      },
+    };
+  }
+};
+
+export const alertingApi = createApi({
+  reducerPath: 'alertingApi',
+  baseQuery: backendSrvBaseQuery(),
+  endpoints: (build) => ({
+    discoverAmFeatures: build.query({
```

Comment:
I think we can use the generics here `build.query<AlertmanagerApiFeatures, { amSourceName: string }>` so we don't need the inline typing